ui/admin/app/components/form/storage-bucket/index.hbs

```diff
@@ -2,455 +2,23 @@
-        </F.HelperText>
-
-        {{#if @model.errors.role_tags}}
-          <F.Error as |E|>
-            {{#each @model.errors.role_tags as |error|}}
-              <E.Message>{{error.message}}</E.Message>
-            {{/each}}
-          </F.Error>
-        {{/if}}
-      </:fieldset>
-      <:field as |F|>
-        <F.TextInput
-          @name='role_tags'
-          @options={{@model.role_tags}}
-          @model={{@model}}
-        />
-      </:field>
-    </Form::Field::ListWrapper>
-  {{/if}}
-
-  <Hds::Form::TextInput::Field
-    @isRequired={{true}}
-    @value={{@model.worker_filter}}
-    @isInvalid={{@model.errors.worker_filter}}
-    @type='text'
-    name='worker_filter'
-    disabled={{form.disabled}}
-    {{on 'input' (set-from-event @model 'worker_filter')}}
-    as |F|
-  >
-    <F.Label>{{t 'resources.storage-bucket.form.worker_filter.label'}}</F.Label>
-    <F.HelperText>
-      {{t 'resources.storage-bucket.form.worker_filter.help'}}
-      <Hds::Link::Inline @href={{doc-url 'storage-bucket.worker-filter'}}>
-        {{t 'actions.learn-more'}}
-      </Hds::Link::Inline>
-    </F.HelperText>
-    {{#if @model.errors.worker_filter}}
-      <F.Error as |E|>
-        {{#each @model.errors.worker_filter as |error|}}
-          <E.Message>{{error.message}}</E.Message>
-        {{/each}}
-      </F.Error>
-    {{/if}}
-  </Hds::Form::TextInput::Field>
-
-  {{#unless this.showDynamicCredentials}}
-    <Hds::Form::Checkbox::Field
-      name='disable_credential_rotation'
-      disabled={{form.disabled}}
-      checked={{@model.disable_credential_rotation}}
-      {{on 'change' this.toggleDisableCredentialRotation}}
-      as |F|
-    >
-      <F.Label>{{t
-          'resources.storage-bucket.form.disable_credential_rotation.label'
-        }}</F.Label>
-      <F.HelperText>
-        {{t 'resources.storage-bucket.form.disable_credential_rotation.help'}}
-        <Hds::Link::Inline
-          @href={{doc-url 'storage-bucket.disable-credential-rotation'}}
-        >
-          {{t 'actions.learn-more'}}
-        </Hds::Link::Inline>
-      </F.HelperText>
-    </Hds::Form::Checkbox::Field>
-  {{/unless}}
-  {{#if (can 'save model' @model)}}
-    <form.actions
-      @enableEditText={{t 'actions.edit-form'}}
-      @submitText={{t 'actions.save'}}
-      @cancelText={{t 'actions.cancel'}}
-    />
-  {{/if}}
-</Rose::Form>
\ No newline at end of file
+{{#if @model.type}}
+  {{component
+    (concat 'form/storage-bucket/' @model.compositeType)
+    model=@model
+    submit=@submit
+    cancel=@cancel
+    updateScope=this.updateScope
+    scopeFieldDescription=this.scopeFieldDescription
+    scopes=@scopes
+    pluginTypes=this.pluginTypes
+    changePluginType=@changePluginType
+    changeCredentialType=@changeCredentialType
+    credentials=this.credentials
+    selectedCredentialType=this.selectedCredentialType
+    updateTypeSelection=this.updateTypeSelection
+    showDynamicCredentials=this.showDynamicCredentials
```

Comment:
Is everything shared by both aws and minIO?

Reply:
@DhariniJeeva 
You are right, the next one, are just need by AWS.

```
credentials=this.credentials
selectedCredentialType=this.selectedCredentialType
updateTypeSelection=this.updateTypeSelection
showDynamicCredentials=this.showDynamicCredentials
```

Reply:
@DhariniJeeva 
TBH I do believe the storage-bucket form itself needs a refactor, I think [this could be simplified](https://github.com/hashicorp/boundary-ui/blob/main/ui/admin/app/components/form/storage-bucket/index.js) and I am ok creating a jira ticket for it.

BUT if you have a solution to not pass these 4 attributes mentioned above that belongs JUST to AWS I will apply such solution now, and leavet the refactor of the component for later. 

Let me know your thoughts 😉

Reply:
I was thinking we could split the JS file and have the AWS-specific actions/getters in the AWS folder since we now have separate folders for each type. Any additional refactoring can be postponed later due to time constraints.